include/envoy/server/instance.h

```diff
@@ -145,6 +145,11 @@ class Instance {
   virtual void shutdownAdmin() PURE;
 
   /**
+   * @return Singleton::Manager& the server-wide singleton manager.
+   */
+  virtual Singleton::Manager& singletonManager() PURE;
```

Comment:
It probably could/should. I basically did this because I did not want to add another top level singleton and plumb it through everywhere. I don't think it's super critical that we back-fill but could open an issue to track.

I think this will be generally useful for things that are added on or not core to the server such that everyone will use no matter what (.e.g, redis, mongo, etc.). Everyone will use the cluster manager, init manager, listener manager, etc. so those are a gray area IMO.